File: app/src/main/java/tiger/com/lp/dailydemo/designpatterns/proxy/UserDao.java
package tiger.com.lp.dailydemo.designpatterns.proxy;

/**
 * @author gxh
 * @email xunhu.gui@ds365.com
 * @date 2018/2/11 0011
 * @description
 */

public class UserDao implements IUserDao{
    @Override
    public void save(int age) {
        System.out.println("我已经" + age + "岁了。");
    }
}

File: app/src/main/java/tiger/com/lp/dailydemo/mvvm/db/UserDao.kt
package tiger.com.lp.dailydemo.mvvm.db

import androidx.lifecycle.LiveData
import androidx.room.*
import tiger.com.lp.dailydemo.mvvm.model.User

/**
 * @author : Xunhu.gui
 * @Email : xunhu.gui@tadu.com
 * @date : 2021/9/30
 * @Description :
 */
@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(user: User)

    @Delete
    fun deleteUser(user: User)

    @Query("select * from user where name = :name")
    fun getUserByName(name: String) : LiveData<User>
}